### ISSUE TITLE
Minimal jax-converted operation fails on provided input fails on WebGL (CPU ok)

### ISSUE THREAD

**Issue by benoitparis** [tensorflow/tfjs#7464]
The code works fine, except if the user wants to provide input **and** use the default WebGL backend.
Setting the backend to cpu makes it work.
The operation is a minimal 1-dim vector normalization. Removing it and just mapping the value makes it work.
Hard-coding the value makes it work.

I think this is a casting issue.

Or maybe I'm misusing the API in a very basic way? I'm pretty sure you guys have working demoes that take inputs.

**Describe the current behavior**

Should log 0.125 on the console like it does for CPU

**Describe the expected behavior**

Logs 0 on the console

Here is the code for reproduction:

Jax converted code:

```
import jax.numpy as jnp
import tensorflow as tf
import tensorflowjs as tfjs

params = {}


def main(ray_params, xs):
    return jnp.linalg.norm(xs)


tfjs.converters.convert_jax(
    main,
    params,
    input_signatures=[tf.TensorSpec((1), tf.float32)],
    model_dir='./bugs/input_bug/'
)

```

HTML code:
```

<html lang="en">
  <head>
    <script src="https://cdn.jsdelivr.net/npm/@tensorflow/tfjs@4.2.0"> </script>
  </head>
  <body>
    <script type="text/javascript">
      let model;
      async function setup() {
        // tf.setBackend('cpu'); // OK
        // tf.setBackend('webgl'); // Default, fails
        model = await tf.loadGraphModel('/bugs/input_bug/model.json');
        console.log(tf.getBackend());
        console.log(tf.env());
        console.log(model);
      }
      async function getOutput() {
        const x = tf.tensor([0.125], [1]);
        let result = model.predict(x);
        synced_result = result.dataSync();
        console.log(synced_result[0]);
      }
      setup().then(getOutput);
    </script>
  </body>
</html>
  
```

model.json:

`{"format": "graph-model", "generatedBy": "2.11.0", "convertedBy": "TensorFlow.js Converter v4.2.0", "signature": {"inputs": {"xs_0": {"name": "xs_0:0", "dtype": "DT_FLOAT", "tensorShape": {"dim": [{"size": "1"}]}}}, "outputs": {"output_0": {"name": "Identity:0", "dtype": "DT_FLOAT", "tensorShape": {}}}}, "modelTopology": {"node": [{"name": "PartitionedCall/jax2tf_main_/pjit_norm_/Sum/reduction_indices", "op": "Const", "attr": {"dtype": {"type": "DT_INT32"}, "value": {"tensor": {"dtype": "DT_INT32", "tensorShape": {"dim": [{"size": "1"}]}}}}}, {"name": "xs_0", "op": "Placeholder", "attr": {"shape": {"shape": {"dim": [{"size": "1"}]}}, "dtype": {"type": "DT_FLOAT"}}}, {"name": "PartitionedCall/jax2tf_main_/pjit_norm_/ArithmeticOptimizer/ReplaceMulWithSquare_Mul", "op": "Square", "input": ["xs_0"], "attr": {"T": {"type": "DT_FLOAT"}}}, {"name": "PartitionedCall/jax2tf_main_/pjit_norm_/Sum", "op": "Sum", "input": ["PartitionedCall/jax2tf_main_/pjit_norm_/ArithmeticOptimizer/ReplaceMulWithSquare_Mul", "PartitionedCall/jax2tf_main_/pjit_norm_/Sum/reduction_indices"], "attr": {"T": {"type": "DT_FLOAT"}, "Tidx": {"type": "DT_INT32"}, "keep_dims": {"b": false}}}, {"name": "PartitionedCall/jax2tf_main_/pjit_norm_/Sqrt", "op": "Sqrt", "input": ["PartitionedCall/jax2tf_main_/pjit_norm_/Sum"], "attr": {"T": {"type": "DT_FLOAT"}}}, {"name": "Identity", "op": "Identity", "input": ["PartitionedCall/jax2tf_main_/pjit_norm_/Sqrt"], "attr": {"T": {"type": "DT_FLOAT"}}}], "library": {}, "versions": {"producer": 1286}}, "weightsManifest": [{"paths": ["group1-shard1of1.bin"], "weights": [{"name": "PartitionedCall/jax2tf_main_/pjit_norm_/Sum/reduction_indices", "shape": [1], "dtype": "int32"}]}]}`

**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow.js): Jax code, very simple
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Windows, 
- Browser version: Chrome Version 110.0.5481.178 (Build officiel) (64 bits)
- Pipfile:
```
[[source]]
url = "https://pypi.org/simple"
verify_ssl = true
name = "pypi"

[packages]
numpy = "==1.23.5"
jaxlib = "==0.4.4"
jax = "==0.4.4"
tensorflow = "==2.11.0"
tensorflowjs = "==4.2.0"

[dev-packages]

[requires]
python_version = "3.10"
```

**Comment by benoitparis** [comment]
Using tensor2d yields the same thing, but in a different way (falls back to CPU automatically instead of putting zeroes in the output):

```
Could not get context for WebGL version 2
tfjs@4.2.0:17 Could not get context for WebGL version 1
tfjs@4.2.0:17 Initialization of backend webgl failed
jx @ tfjs@4.2.0:17
t.initializeBackend @ tfjs@4.2.0:17
t.initializeBackendsAndReturnBest @ tfjs@4.2.0:17
get @ tfjs@4.2.0:17
t.makeTensor @ tfjs@4.2.0:17
Pk @ tfjs@4.2.0:17
Bk @ tfjs@4.2.0:17
Gk @ tfjs@4.2.0:17
t.loadSync @ tfjs@4.2.0:17
(anonymous) @ tfjs@4.2.0:17
Promise.then (async)
t.load @ tfjs@4.2.0:17
(anonymous) @ tfjs@4.2.0:17
l @ tfjs@4.2.0:17
(anonymous) @ tfjs@4.2.0:17
(anonymous) @ tfjs@4.2.0:17
Nv @ tfjs@4.2.0:17
o @ tfjs@4.2.0:17
(anonymous) @ tfjs@4.2.0:17
(anonymous) @ tfjs@4.2.0:17
SV @ tfjs@4.2.0:17
e.loadGraphModel @ tfjs@4.2.0:17
setup @ index.html:12
(anonymous) @ index.html:23
tfjs@4.2.0:17 Error: WebGL is not supported on this device
    at new t (tfjs@4.2.0:17:1051692)
    at Object.factory (tfjs@4.2.0:17:1075787)
    at t.initializeBackend (tfjs@4.2.0:17:164511)
    at t.initializeBackendsAndReturnBest (tfjs@4.2.0:17:165909)
    at e.get (tfjs@4.2.0:17:180704)
    at t.makeTensor (tfjs@4.2.0:17:170627)
    at Pk (tfjs@4.2.0:17:188873)
    at Bk (tfjs@4.2.0:17:188917)
    at Object.Gk [as decodeWeights] (tfjs@4.2.0:17:192341)
    at t.loadSync (tfjs@4.2.0:17:792663)
```

Using as input:


`        const x = tf.tensor2d([0.125], [1, 1]);`

With that spec:


`    input_signatures=[tf.TensorSpec((1, 1), tf.float32)],`

**Comment by benoitparis** [comment]
Welp, the [LiT demo here](https://google-research.github.io/vision_transformer/lit/) also tells me 

```
Could not get context for WebGL version 2
Could not get context for WebGL version 1
Initialization of backend webgl failed
Error: WebGL is not supported on this device
```

I have no problems displaying other WebGL otherwise, I have no problems displaying any example at https://threejs.org/examples/ for instance.

Is this a device-specific bug? (XPS15 with NVIDIA GTX 960M here)

How can I provide other info? Does the LiT demo work well on other devices?

**Comment by pyu10055** [comment]
The demo works in my latest chome on a linux box, what os and browser are you using?

**Comment by gaikwadrahul8** [comment]
Hi, @benoitparis 

Could you please help us with information asked by @pyu10055 in previous comment so we'll try to replicate the same issue from our end ? Thank you!
